pkg/jmxfetch/jmxfetch.go

```diff
@@ -142,7 +143,27 @@ func (j *JMXFetch) Start(manage bool) error {
 
 	// Specify a maximum memory allocation pool for the JVM
 	javaOptions := j.JavaOptions
-	if config.Datadog.GetBool("jmx_use_cgroup_memory_limit") {
+
+	incompatibleOptions := false
+	useContainerSupport := config.Datadog.GetBool("jmx_use_container_support")
+	useCgroupMemoryLimit := config.Datadog.GetBool("jmx_use_cgroup_memory_limit")
+
+	if useContainerSupport && useCgroupMemoryLimit {
+		log.Warnf("Skipping incompatible options %q and %q, only one should be set at a time", jvmContainerSupport, jvmCgroupMemoryAwareness)
+		incompatibleOptions = true
+	}
+	if incompatibleOptions || !(useContainerSupport || useCgroupMemoryLimit) {
```

Comment:
Honest question: shouldn't we just fail starting here instead of picking a value ourselves?

Reply:
I wanted to follow the logic of non-failure on incompatible JVM options as it was already done for `jmx_use_container_support` but that may be a bit too much there. I'm down for that change

Reply:
if I were the maintainer of this I would prefer a failure here as it gets tricky, but I'm not and maybe this logic is actually fine as long as we get the runtime config in the flare. @olivielpeau what do you say?

Reply:
I feel like if this doesn't fail, many users will happily go about their lives while having an _"invalid/unexpected"_ configuration. I think we should probably not launch `JMXFetch` in this scenario and log a big fat error. I feel that's probably a preferable behavior.